optimum/exporters/onnx/config.py

```diff
@@ -66,6 +69,59 @@ def inputs(self) -> Dict[str, Dict[int, str]]:
             }
         return common_inputs
 
+    def post_process_exported_models(self, path: Path, models_and_onnx_configs, onnx_files_subpaths):
+        # Attempt to merge only if the decoder-only was exported separately without/with past
+        if self.use_past is True and len(models_and_onnx_configs) == 2:
+            if onnx_files_subpaths is not None:
+                decoder_path = Path(path, onnx_files_subpaths[0])
+                decoder_with_past_path = Path(path, onnx_files_subpaths[1])
+            else:
+                decoder_path = Path(path, ONNX_DECODER_NAME + ".onnx")
+                decoder_with_past_path = Path(path, ONNX_DECODER_WITH_PAST_NAME + ".onnx")
+            decoder_merged_path = Path(path, ONNX_DECODER_MERGED_NAME + ".onnx")
+            try:
+                merge_decoders(
+                    decoder=decoder_path,
+                    decoder_with_past=decoder_with_past_path,
+                    save_path=decoder_merged_path,
+                )
+            except Exception as e:
+                raise Exception(f"Unable to merge decoders. Detailed error: {e}")
+            os.remove(decoder_path)
+            os.remove(decoder_with_past_path)
+
+            # In order to do the validation of the two branches on the same file
+            onnx_files_subpaths = [decoder_merged_path.name, decoder_merged_path.name]
+
+            # We validate the two branches of the decoder model then
+            models_and_onnx_configs[ONNX_DECODER_NAME][1].is_merged = True
+            models_and_onnx_configs[ONNX_DECODER_NAME][1].use_cache_branch = False
+
+            # Past key values won't be generated by default, but added in the input
+            models_and_onnx_configs[ONNX_DECODER_NAME][1].use_past = False
+            models_and_onnx_configs[ONNX_DECODER_NAME][1].use_past_in_inputs = True
+
+            models_and_onnx_configs[ONNX_DECODER_WITH_PAST_NAME][1].use_cache_branch = True
+            models_and_onnx_configs[ONNX_DECODER_WITH_PAST_NAME][1].is_merged = True
+
+        return models_and_onnx_configs, onnx_files_subpaths
+
+    def generate_dummy_inputs_for_validation(self, reference_model_inputs: Mapping[str, Any]) -> Mapping[str, Any]:
+        if self.is_merged is True and self.use_cache_branch is True:
+            reference_model_inputs["use_cache_branch"] = DummyInputGenerator.constant_tensor(shape=[1], value=True)
+        elif self.is_merged is True and self.use_cache_branch is False:
+            reference_model_inputs["use_cache_branch"] = DummyInputGenerator.constant_tensor(shape=[1], value=False)
+
+            # We don't support optional inputs for now, so even though the non-cache branch is used,
```

Comment:
Ok, so you are reusing previous dummy method in the onnx config? Can we just remove the `use_cache_branch` in conditional statement in the onnx config and reuse the dummy pkv that it generate and add only the specific input `se_cache_branch` here?

Reply:
No, I don't think this is possible. The issue is that the dummy inputs are passed first to the PyTorch model for validation, see: https://github.com/huggingface/optimum/blob/c86ca5666cf3934ae6242dfbf91fd7d4d45eb0f2/optimum/exporters/onnx/convert.py#L186-L237

This means that the input `past_key_values` should **not** be passed to the PyTorch model when we want to validate the non-cache branch. Therefore, adding the dummy past key values is done only in `generate_dummy_inputs_for_validation` (that possibly edits the inputs for ORT inference) and not `generate_dummy_inputs`, as these inputs are only required for ORT and not PyTorch.

Reply:
OK, I SEE